config/convert.go

```diff
@@ -222,6 +224,64 @@ func ConvertAs2_0_0(in Config) (types.Config, error) {
 		})
 	}
 
+	// etcd and etcd2 get drop in files with the configuration
+	if !isZero(in.Etcd) {
+		out.Systemd.Units = append(out.Systemd.Units, types.SystemdUnit{
+			Name:   "etcd.service",
+			Enable: true,
+			DropIns: []types.SystemdUnitDropIn{{
+				Name:     "20-ignition-etcd.conf",
+				Contents: serviceContents(in.Etcd),
+			}},
+		})
+	}
+
+	if !isZero(in.Etcd2) {
+		out.Systemd.Units = append(out.Systemd.Units, types.SystemdUnit{
+			Name:   "etcd2.service",
+			Enable: true,
+			DropIns: []types.SystemdUnitDropIn{{
+				Name:     "20-ignition-etcd2.conf",
+				Contents: serviceContents(in.Etcd2),
+			}},
+		})
+	}
+
+	// flannel gets a /run/flannel/options.env" file with the configuration
+	if !isZero(in.Flannel) {
+		envVars := strings.Join(getEnvVars(in.Flannel), "\n")
+		if envVars != "" {
+			// Since the files go in a tmpfs, we need to tell ignition about the /run filesystem
+			// XXX How do we ensure __runTmpfs doesn't interfere with user supplied filesystem names?
```

Comment:
Well, you do know all of the filesystems provided by the user. You could just double check that the name doesn't collide.